libs/header/p2p/exchange.go

```diff
@@ -220,13 +225,21 @@ func (ex *Exchange[H]) performRequest(
 		return make([]H, 0), nil
 	}
 
-	if len(ex.trustedPeers) == 0 {
-		return nil, fmt.Errorf("no trusted peers")
+	for {
+		//nolint:gosec // G404: Use of weak random number generator
+		idx := rand.Intn(len(ex.trustedPeers))
```

Comment:
Is it intended that errored peer could be retried before all non-tried peers has been attempted? For example it could be the same random result subsequently.

Reply:
Yeah, but it's fine. The trustedPeers are assumed to be highly available, so retrying the same peer is fine. Also, this is not the final solution for retrying and peer selection logic: #1665